src/EventListener/InsertTagsListener.php

```diff
@@ -48,10 +48,13 @@ public function __construct(ContaoFrameworkInterface $framework)
      * Replaces news insert tags.
      *
      * @param string $tag
+     * @param bool   $useCache
+     * @param mixed  $cacheValue
+     * @param array  $flags
      *
      * @return string|false
      */
-    public function onReplaceInsertTags(string $tag)
+    public function onReplaceInsertTags(string $tag, bool $useCache, $cacheValue, array $flags)
```

Comment:
This cannot be right. First, both `$useCache` and `$cacheValue` are not used anywhere, so why did you add them? Second, if you add new arguments to an existing method, you should make them optional. If you cannot make them optional (which is not the case here), you have to adjust the `listeners.yml` file accordingly.

Reply:
I have added the `$useCache` and `$cacheValue` to get the access to the `$flags`. Should I use `func_get_arg()` instead?

Reply:
The `core-bundle` passes the arguments in that order: https://github.com/contao/core-bundle/blob/master/src/Resources/contao/library/Contao/InsertTags.php#L953

Reply:
Ah, now I understand. Then its fine I guess.

Reply:
But the new arguments should be optional indeed or otherwise it is a BC break – even if this method is used only by the hook. Should I change that?

Reply:
Yes, I think using `func_get_arg()` is the best solution in this case.

Reply:
Sorry, by optional I meant providing their default values:

```php
public function onReplaceInsertTags(string $tag, bool $useCache = true, $cacheValue = null, array $flags = [])
```

Reply:
Symfony does a lot of `func_num_args() > 4` … check to see if the argument is there and then use it.

Reply:
How about this? 599ea6d

Reply:
No, Symfony only does this for deprecated stuff, e.g.

https://github.com/symfony/symfony/blob/8cd2193a82171686f41fd073d2157de0293e3bb2/src/Symfony/Component/Yaml/Yaml.php#L93-L107

But never for regular method calls.

Reply:
So let's use the "normal" approach in 4fb9357.